co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstanceManager.java

```diff
@@ -135,24 +151,79 @@ public V2xMessageTransmission onV2XMessageTx(InetAddress sourceAddr, CarmaV2xMes
+     * Callback to be invoked when an RSU receives a V2X Message from the NS-3
+     * simulation
+     * 
+     * @param rxMsg   The V2X Message received
      * @param rxRsuId The Host ID of the vehicle receiving the data
-     * @throws RuntimeException If the socket used to communicate with the platform experiences failure
+     * @throws RuntimeException If the socket used to communicate with the platform
+     *                          experiences failure
      */
     public void onV2XMessageRx(byte[] rxMsg, String rxRsuId) {
-        if (!managedInstances.containsKey(rxRsuId))  {
+        if (!managedInstances.containsKey(rxRsuId)) {
             return;
         }
 
         InfrastructureInstance rsu = managedInstances.get(rxRsuId);
         try {
-            rsu.sendMsgs(rxMsg);
+            rsu.sendV2xMsg(rxMsg);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
+    /**
+     * Callback to be invoked when a infrastructure instance receives a simulated
+     * object detection from a registered simulated sensors. Only send object detection
+     * to the infrastructure instance that contains the sensor that reported it.
+     * 
+     * @param detection Detected Object.
+     */
+    public void onDetectedObject(DetectedObject detection) {
+        for (InfrastructureInstance instance : managedInstances.values()) {
+            if (instance.containsSensor(detection.getSensorId())) {
+                try {
+                    instance.sendInteraction(encodeMsg(detection));
+                    // Assuming each sensor would only ever be registered to a single infrastructure
+                    // instance
+                    break;
+                } catch (IOException e) {
+                    log.error("Error occured:  {}", e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Helper method to serialize message into JSON and encode as bytes.
+     * 
+     * @param message java object containing message information
+     * @return bytes encoded from JSON string representation of object.
+     */
+    private byte[] encodeMsg(Object message) {
+        Gson gson = new Gson();
+        return gson.toJson(message).getBytes();
+    }
+
+    
+    /**
+     * This function is used to send out encoded timestep update to all registered
+     * instances the manager has on the managed instances map
+     * 
+     * @param message This time message is used to store current seq and timestep
+     *                from the ambassador side
+     * @throws IOException
+     */
+    public void onTimeStepUpdate(InfrastructureTimeMessage message) throws IOException {
+        if (managedInstances.size() == 0) {
+            log.debug("There are no registered instances");
+        }
+
+        for (InfrastructureInstance currentInstance : managedInstances.values()) {
+            currentInstance.sendTimeSyncMsg(encodeMsg(message));
```

Comment:
This is somewhere I think we could just make `sendTimeSyncMsg` consume InfrastructureTimeMessage like I noted prior, then the entire encodeMsg function is no longer needed (or at least not needed here) and then I think the API is a little clearer in terms of intended use.